filecoin/store/receipt.js

```diff
@@ -0,0 +1,131 @@
+import {
+  S3Client,
+  GetObjectCommand,
+  ListObjectsV2Command
+} from '@aws-sdk/client-s3'
+import * as CAR from '@ucanto/transport/car'
+import { StoreOperationFailed, RecordNotFound } from '@web3-storage/filecoin-api/errors'
+
+/**
+ * Abstraction layer with Factory to perform operations on bucket storing
+ * handled receipts.
+ *
+ * @param {string} region
+ * @param {string} invocationBucketName
+ * @param {string} workflowBucketName
+ * @param {import('@aws-sdk/client-s3').ServiceInputTypes} [options]
+ */
+export function createReceiptStore(region, invocationBucketName, workflowBucketName, options = {}) {
+  const s3client = new S3Client({
+    region,
+    ...options,
+  })
+  return useReceiptStore(s3client, invocationBucketName, workflowBucketName)
+}
+
+/**
+ * @param {S3Client} s3client
+ * @param {string} invocationBucketName
+ * @param {string} workflowBucketName
+ * @returns {import('@web3-storage/filecoin-api/storefront/api').ReceiptStore}
+ */
+export const useReceiptStore = (s3client, invocationBucketName, workflowBucketName) => {
+  return {
+    put: async (record) => {
+      return {
+        error: new StoreOperationFailed('no new receipt should be put by storefront')
```

Comment:
...then why does the `ReceiptStore` type require it? I had the same problem with billing, you could either use `Omit` or have multiple store types that you combine for different requirements e.g. https://github.com/web3-storage/w3infra/blob/72a628f2e3ae20cd04ed81391a2ba5959c6d6def/billing/lib/api.ts#L294-L310

Not blocking!

Reply:
We need PUT in the tests interface, I agree it is not ideal. I will fill in an issue to make it specific for src and test

Reply:
Ok cool. I defined extended test stores for put/get etc. https://github.com/web3-storage/w3infra/blob/72a628f2e3ae20cd04ed81391a2ba5959c6d6def/billing/test/lib/api.ts#L25